dynamorm/model.py

```diff
@@ -283,15 +283,24 @@ def scan(cls, scan_kwargs=None, **kwargs):
             Thing.scan(address__state="CA")
             Thing.scan(address__state__begins_with="C")
 
+        This returns a generator, which will continue to yield items until all matching the scan are produced,
+        abstracting away pagination. More information on scan pagination: http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Scan.html#Scan.Pagination
+
         :param dict scan_kwargs: Extra parameters that should be passed through to the Table scan function
         :param \*\*kwargs: The key(s) and value(s) to filter based on
         """  # noqa
         resp = cls.Table.scan(scan_kwargs=scan_kwargs, **kwargs)
-        return [
-            cls.new_from_raw(raw)
-            for raw in resp['Items']
-            if raw is not None
-        ]
+        for raw in resp['Items']:
+            if raw is not None:
+                yield cls.new_from_raw(raw)
+
+        while 'LastEvaluatedKey' in resp:
+            scan_kwargs = scan_kwargs or {}
+            scan_kwargs['ExclusiveStartKey'] = resp['LastEvaluatedKey']
+            resp = cls.Table.scan(scan_kwargs=scan_kwargs, **kwargs)
+            for raw in resp['Items']:
+                if raw is not None:
+                    yield cls.new_from_raw(raw)
```

Comment:
Throwing this all into a separate helper (had it in one originally but found myself increasing scope)

Since we don't offer much different re: functionality sugar on top of query vs scan, I was thinking to just make query a generator as well. Thoughts?